scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -832,3 +832,117 @@ def tearDown(self):
+            self.admin.assert_icommand(['itouch', 'issue_7909'], 'STDERR')
+
+            # Show the AVU containing the error exists on the session collection.
+            expected_output = [
+                f'attribute: {attr_name}\n',
+                'value: Invalid value for [parser]. Expected Parser.GENQUERY1 or Parser.GENQUERY2.\n'
+            ]
+            self.admin.assert_icommand(['imeta', 'ls', '-C', self.admin.session_collection], 'STDOUT', expected_output)
+
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-python', 'Requires PREP.')
+    def test_genquery_iterator_supports_genquery2__issue_7909(self):
+        rule_file = f'{self.admin.local_session_dir}/test_genquery_iterator_supports_genquery2__issue_7909.r'
+
+        with open(rule_file, 'w') as rf:
+            rf.write(dedent(f'''
+            def main(rule_args, callback, rei):
+                from genquery import Query, Parser
+                for r in Query(callback, 'COLL_NAME', "COLL_NAME = '{self.admin.session_collection}'", parser=Parser.GENQUERY2):
+                    callback.writeLine('stdout', f'row: {{r}}')
+            INPUT null
+            OUTPUT ruleExecOut
+            '''))
+
+        # Execute the rule.
+        rep_instance = IrodsConfig().default_rule_engine_plugin + '-instance'
+        expected_output = [f"row: ['{self.admin.session_collection}']"]
+        self.admin.assert_icommand(['irule', '-r', rep_instance, '-F', rule_file], 'STDOUT', expected_output)
+
+    @unittest.skipUnless(plugin_name == 'irods_rule_engine_plugin-python', 'Requires PREP.')
+    def test_genquery_iterator_supports_genquery2_offset_and_limit__issue_7909(self):
+        # Create some data objects.
+        data_name_prefix = 'test_genquery_iterator_supports_genquery2_offset_and_limit__issue_7909'
+        for i in range(10):
+            self.admin.assert_icommand(['itouch', f'{data_name_prefix}.{i}'])
+
+        # Test offset functionality.
+        rule_file = f'{self.admin.local_session_dir}/test_genquery_iterator_supports_genquery2_offset_and_limit__issue_7909.r'
+        with open(rule_file, 'w') as rf:
+            rf.write(dedent(f'''
+            def main(rule_args, callback, rei):
+                from genquery import Query, Parser
+                rows = []
+                for r in Query(callback, 'DATA_NAME', "COLL_NAME = '{self.admin.session_collection}' and DATA_NAME like '{data_name_prefix}.%'", offset=3, parser=Parser.GENQUERY2):
+                    rows.append(r)
+                callback.writeLine('stdout', f'count: {{len(rows)}}')
+            INPUT null
+            OUTPUT ruleExecOut
+            '''))
+
+        # Execute the rule.
+        rep_instance = IrodsConfig().default_rule_engine_plugin + '-instance'
+        expected_output = ['count: 7']
+        self.admin.assert_icommand(['irule', '-r', rep_instance, '-F', rule_file], 'STDOUT', expected_output)
+
+        # Test limit functionality.
```

Comment:
I feel like it would be cleaner to split these tests up, but having them combined is also fine.

Reply:
Oh, you're talking about separating the testing of `limit` and `offset`?

No problem. Will separate.

Reply:
Separated them into two distinct tests.

Reply:
Yes, sorry, I should have clarified. Thanks